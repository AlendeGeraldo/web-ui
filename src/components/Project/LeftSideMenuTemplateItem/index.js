import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Icon } from 'antd';

import { setTemplate } from '../../../store/actions/experimentActions';

/**
 * Common project menu template item.
 */
const LeftSideMenuTemplateItem = ({
  disabled = false,
  experimentId = null,
  projectId,
  onSetTemplate,
  template,
}) => (
  // menu item container
  <div
    onClick={() => {
      onSetTemplate(projectId, experimentId, template);
    }}
    className={`collapse-menu-items${template.disabled ? ' disabled' : ''}${
      disabled ? ' disabled' : ''
    }`}
    role='presentation'
  >
    {/* menu item icon */}
    <Icon className='icon-collapse-header' type='more' />

    {/* menu item name */}
    {template.name}
  </div>
);

const mapStateToProps = (state) => {
  return {
    projectId: state.experiment.projectId,
    experimentId: state.experiment.uuid,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    onSetTemplate: (projectId, experimentId, template) =>
      dispatch(setTemplate(projectId, experimentId, template)),
  };
};

LeftSideMenuTemplateItem.propTypes = {
  /** item is disabled */
  disabled: PropTypes.bool,
  /** current experiment id */
  experimentId: PropTypes.string,
  /** current project id */
  projectId: PropTypes.string,
  /** on set template handle function */
  onSetTemplate: PropTypes.func.isRequired,
  /** item template */
  template: PropTypes.objectOf(PropTypes.any).isRequired,
};

LeftSideMenuTemplateItem.defaultProps = {
  /** disabled prop default is null */
  disabled: false,
  /** current experiment id default is null */
  experimentId: null,
  /** current project id default is null */
  projectId: null,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LeftSideMenuTemplateItem);
