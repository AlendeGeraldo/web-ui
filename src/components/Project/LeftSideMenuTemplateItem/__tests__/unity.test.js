import React from 'react';
import { shallow } from 'enzyme';
import LeftSideMenuTemplateItem from '..';

const handleClick = jest.fn();

describe('LeftSideMenuTemplateItem component', () => {
  it('is expected render without crashing', () => {
    shallow(
      <LeftSideMenuTemplateItem
        handleClick={handleClick}
        template={{ disabled: false }}
        disabled={false}
      />
    );
  });

  it('is expected render html correctly', () => {
    const wrapper = shallow(
      <LeftSideMenuTemplateItem
        handleClick={handleClick}
        template={{ disabled: true }}
        disabled
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('onClick', () => {
    const wrapper = shallow(
      <LeftSideMenuTemplateItem
        handleClick={handleClick}
        template={{ disabled: false }}
      />
    );
    const div = wrapper.find('div');
    div.simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });
});
