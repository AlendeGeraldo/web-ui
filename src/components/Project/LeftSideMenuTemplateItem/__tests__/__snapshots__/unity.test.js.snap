// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftSideMenuTemplateItem component is expected render html correctly 1`] = `
<div
  className="collapse-menu-items disabled disabled"
  onClick={[Function]}
  role="presentation"
>
  <Icon
    className="icon-collapse-header"
    type="more"
  />
</div>
`;
